packages/docs-reanimated/docs/reanimated-babel-plugin/about.md

```diff
@@ -0,0 +1,217 @@
+
+A worklet is:
+
+- A function that contains a `'worklet'` directive at its very top, i.e.:
+
+```ts
+function foo() {
+  'worklet';
+  console.log('Hello from worklet');
+}
+```
+
+- A function that is _autoworkletizable_, i.e.:
+
+```ts
+useAnimatedStyle(() => {
+  // This function will be ran on the Worklet Runtime,
+  // hence it's in a workletizable context and will be
+  // autoworkletized. You don't need to add the 'worklet' directive here.
+  return {
+    width: 100,
+  };
+});
+```
+
+## What can be a worklet?
+
+Currently, Reanimated Babel Plugin supports the following constructs as worklets:
+
+- Arrow Function Expressions
+
+```ts
+const foo = () => {
+  'worklet';
+  console.log('Hello from ArrowFunctionExpression');
+};
+```
+
+- Function Declarations
+
+```ts
+function foo() {
+  'worklet';
+  console.log('Hello from FunctionDeclaration');
+}
+```
+
+- Function Expressions
+
+```ts
+const foo = function () {
+  'worklet';
+  console.log('Hello from FunctionExpression');
+};
+```
+
+- Object Methods
+
+```ts
+const obj = {
+  foo() {
+    'worklet';
+    console.log('Hello from ObjectMethod');
+  },
+};
+```
+
+## Autoworkletization
+
+To reduce boilerplate code and relieve the user from knowing the internals of Reanimated, Reanimated Babel Plugin tries to detect automatically whether a function should be workletized. For example, all callbacks in Reanimated API that are ran on the Worklet runtime are autoworkletized, such as:
+
+- `useAnimatedStyle`
+- `useAnimatedProps`
+- `runOnUI`
+
+etc.
+
+Thanks to that, you don't need to add the `'worklet'` directive to your code when you're not exploring any advanced use cases.
+
+### Referencing worklets
+
+You can define worklets before they are used, and the plugin will autoworkletize them too:
+
+```ts
+function foo() {
+  // You don't need to mark it as a worklet.
+  return { width: 100 };
+}
+
+const style = useAnimatedStyle(foo); // You don't need to define an inline function here.
+```
+
+### Objects aggregating worklets
+
+In some APIs, like `useAnimatedScrollHandler` you can pass an object that contains worklets instead of a function:
+
+```ts
+const handlerObject = {
+  // You don't need to mark these methods as worklets.
+  onBeginDrag() {
+    console.log('Dragging...');
+  },
+  onScroll() {
+    console.log('Scrolling...');
+  },
+};
+
+const handler = useAnimatedScrollHandler(handlerObject);
+```
+
+### Workletizing whole files (experimental)
+
+You can mark a file as a workletizable file by adding the `'worklet'` directive to the top of the file. This will make all _top-level_ entities workletized. This can come in handy for files that contain multiple worklets.
+
+```ts
+'worklet';
+
+function foo() {
+  // This function will be autoworkletized.
+  return { width: 100 };
+}
+
+function bar() {
+  // This function will be autoworkletized.
+  function foobar() {
+    // This function won't since it's not defined in top-level scope.
+    console.log('Hello from worklet');
```

Comment:
the comments in this snippet and the console log are confusing

normally when we comment functions we put a comment before the name
```js
// this function does something
function superFun() {
``` 

and the console log says `"Hello from worklet"` where the comment says the function won't get workletized

Reply:
Some of the comments I put where the user would expect the `'worklet'` directive to be to get the "wow effect" 😄 However it might make more sense to put all such comments before the functions. What do you think?

Reply:
I changed the ambiguous fragments so now they explicitly say what they refer to.